src/netius/test/base/tls.py

```
@@ -62,6 +124,37 @@ def test_match_hostname(self):
+        certificate = dict(
+            subject=((("organizationName", "Hive"), ("commonName", "domain.com")),),
+            version=3,
+        )
+
+        # with no alternative name in the certificate the common name of
+        # the subject is the one that answers for the host, the entries
+        # of it that are not a common name being skipped
+        self.assertEqual(netius.match_hostname(certificate, "domain.com"), None)
+        self.assertRaises(
+            BaseException, lambda: netius.match_hostname(certificate, "other.com")
+        )
```

Ruff (0.16.2)

[warning] Do not assert blind exception: `BaseException`

(B017)